src/battle_util.c

```diff
@@ -2309,7 +2309,7 @@ u8 AtkCanceller_UnableToUseMove(void)
                 gBattleMons[gBattlerAttacker].status2--;
                 if (gBattleMons[gBattlerAttacker].status2 & STATUS2_CONFUSION)
                 {
-                    if (Random() & 1)
+                    if (Random() & ((B_CONFUSION_SELF_DMG_CHANCE >= GEN_7) ? 2 : 1)
```

Comment:
should be changed to 
if (Random() % ((B_CONFUSION_SELF_DMG_CHANCE >= GEN_7) ? 3: 2 == 0)

Reply:
It's missing one parenthesis also.